newsrevenuehub/rev-engine
# refs #NRH-250 Refactors metadata generation

#### What's this PR do?
Refactors the app to use first_name, last_name, and the expected `mailing` prefixes 
Adds some logic to make sure that the correct metadata is present on the Processor Object.

#### How should this be manually tested?

1. Make a one-time contribution. 

- Any metadata with a ProcessorObject of ALL or PAYMENT should be found on the PaymentIntent Object in Stripe.

2. Make a recurring donation.

- Any metadata with a ProcessorObject of ALL or CUSTOMER should be found on the Customer Object.
- Any metadata with a ProcessorObject of ALL or PAYMENT should be found on the Subscription Object

No other extraneous metadata should be found on any objects.

*NOTE* Customer creation for one-time payments still has not been implemented, so metadata will not be on those objects for one-time payments.

#### Do any changes need to be made before deployment to staging? production? (adding environment variables, for example)?

I have a table dump of my local ContributionMetadata that I will add to `rev-engine-test`. This should give us a common reference to understand future requests or changes for this feature.

For BAD_ACTOR this PR changes the keys sent from `given_name` to `first_name` and `family_name` to `last_name`
